tools/gitpod/Dockerfile

```diff
@@ -0,0 +1,20 @@
+FROM gitpod/workspace-full
+
+RUN sudo apt-get update && \
+    sudo apt-get install --no-install-recommends -y ccache gfortran && \
+    sudo rm -rf /var/lib/apt/lists/*
```

Comment:
Could we use conda like SciPy does? It's much easier to get a complete dev environment that way. Can use the `environment.yml` in the root of the repo for it.

Reply:
I updated the PR to use `environment.yml`. The extra dependencies does increase the size of the  docker image, which makes the "Pulling docker image" screen take around ~1 minute.

Reply:
I think that's still worth it, missing dependencies that are necessary to work on docs or type annotations for example would be worse, because those are tasks that are good for new contributors. And a dev env without IPython is seriously incomplete imho:)